move/abi/sources/abi.move

```diff
@@ -30,57 +60,47 @@ module abi::abi {
         }
     }
 
+    /// Creates a new `AbiWriter` that can fit up to length bytes before to
+    /// overflows.
     public fun new_writer(length: u64): AbiWriter {
-        let mut bytes = vector[];
-        let mut i = 0;
-
-        while (i < 32 * length) {
-            bytes.push_back(0);
-            i = i + 1;
-        };
-
         AbiWriter {
-            bytes,
+            bytes: vector::tabulate!(U256_BYTES * length, |_| 0),
             pos: 0,
         }
     }
 
+    /// Retrieve the bytes from an `AbiWriter`.
     public fun into_bytes(self: AbiWriter): vector<u8> {
         let AbiWriter { bytes, pos: _ } = self;
 
         bytes
     }
 
-    // TODO: check that all bytes were decoded
-    public fun into_remaining_bytes(self: AbiReader): vector<u8> {
-        let AbiReader { bytes, head: _, pos: _ } = self;
-
-        bytes
-    }
-
+    /// Read a `u256` from the next slot of the `AbiReader`. Should be used to read
+    /// other fixed length types as well.
     public fun read_u256(self: &mut AbiReader): u256 {
         let mut var = 0u256;
-        let mut i = 0;
         let pos = self.pos;
 
-        while (i < 32) {
-            var = (var << 8) | (self.bytes[i + pos] as u256);
-            i = i + 1;
-        };
+        U256_BYTES.do!(|i| var = (var << 8) | (self.bytes[i + pos] as u256));
 
-        self.pos = pos + 32;
+        self.pos = pos + U256_BYTES;
 
         var
     }
 
+    /// Wrapper for `read_u256` that casts the result into a `u8`.
     public fun read_u8(self: &mut AbiReader): u8 {
         self.read_u256() as u8
     }
 
+    /// Used to ignore the next variable in an `AbiReader`.
     public fun skip_slot(self: &mut AbiReader) {
-        self.pos = self.pos + 32;
+        self.pos = self.pos + U256_BYTES;
     }
 
+    /// Reads a variable length variable from an `AbiReader`, as bytes. Can be
+    /// converted to other variable length variables as well (such as `Strings`).
```

Comment:
```suggestion
    /// Reads a variable length bytes from an `AbiReader`.
    /// This can be used to read other variable length types, such as `String`.
```